bindings/java/src/org/sleuthkit/datamodel/Blackboard.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Sleuth Kit Data Model
+ *
+ * Copyright 2018 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *	 http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.datamodel;
+
+import java.io.Closeable;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A representation of the blackboard, a place where artifacts and their
+ * attributes are posted.
+ *
+ */
+public final class Blackboard implements Closeable {
+	private SleuthkitCase caseDb;
+
+	/**
+	 * Constructs a representation of the blackboard, a place where artifacts
+	 * and their attributes are posted.
+	 *
+	 * @param casedb The case database.
+	 */
+	public Blackboard(SleuthkitCase casedb) {
+		this.caseDb = casedb;
+	}
+
+	/**
+	 * Closes the blackboard.
+	 *
+	 */
+	@Override
+	public synchronized void close() {
+		caseDb = null;
+	}
+
+	
+	/**
+	 * Gets the list of all unique artifact IDs in use for the given data source
+	 * Gets both static and dynamic IDs.
+	 *
+	 * @param ds_obj_id  data source obj id 
+	 * 
+	 * @return The list of unique IDs
+	 *
+	 * @throws TskCoreException exception thrown if a critical error occurred
+	 *                          within tsk core
+	 */
+	public List<BlackboardArtifact.Type> getArtifactTypesInUseByDataSource(long ds_obj_id) throws TskCoreException {
```

Comment:
This method returns type objects, not ids, so the doc comment is incorrect. Also, I think "static and dynamic ids" is confusing; what is meant is standard and custom types.

We should use Java naming conventions for method parameters: ds_obj_id => dataSourceObjId. This applies to several methods on this class.

We could probably shorten the method name, allowing the dataSourceObjId parameter to distinguish overloads of a getArtifactTypesInUse when we move SleuthkitCase.getArtifactTypesInUse into this class.